readMe.md
# codeact

### 介绍

- 执行命令的时候会在项目根目录添加 code-act-service 目录，并且自动安装依赖并启动；

- 通过 code-act-service 的交互，来实时添加、编辑项目代码，后期目标通过网页配置方式，直接在项目生成各模块的代码到对应位置。

### 安装

npm install -g codeact

### 使用

- 1、在项目根目录执行 codeact，第一次默认会在根目录生成 codeact.config.js 配置文件，后续可以直接修改配置做定制化功能

- 2、可以通过 codeact init 重新初始化 code-act-service 文件

#### 命令说明

| 短名字 | 长名字   | 介绍                                                                             |
| ------ | -------- | -------------------------------------------------------------------------------- |
| -V     | --vision | 查看版本号                                                                       |
| -h     | --help   | 查看帮助文档                                                                     |
| -f     | --find   | 查看属性，可查询参数（gitAddress、serviceFolderName、serviceStop、serviceStart） |
| init   |          | 重新初始化                                                                       |

## Code Act Service 说明

![md_1](https://github.com/aisriver/codeact/blob/master/docs/images/01.png?raw=true)

![md_1](https://github.com/aisriver/codeact/blob/master/docs/images/02.png?raw=true)

配合命令行工具 codeact 使用的一项辅助开发的解决方案

## 文件操作类接口说明

- baseUrl http://localhost:7001

- Folder 相关

| 地址              | 类型 |                            参数 |                     说明 |
| :---------------- | :--: | ------------------------------: | -----------------------: |
| /folder/add       | GET  |                path、folderName |               新增文件夹 |
| /folder/delete    | GET  |                path、folderName |               删除文件夹 |
| /folder/rename    | GET  | path、folderName、newFolderName |             文件夹重命名 |
| /folder/structure | GET  |                                 | 获取工程下的文件结构数据 |

- File 相关

| 地址         | 类型 |           参数 |         说明 |
| :----------- | :--: | -------------: | -----------: |
| /file/add    | GET  | path、fileName |     新增文件 |
| /file/delete | POST |           path |     删除文件 |
| /file/read   | GET  |           path | 文件内容读取 |
| /file/modify | GET  |     path、text | 文件内容修改 |

### 拓展及说明

- codeact 会生成配置文件 你可以自己搭建一个服务关联到 codeact，一切都可以自定义

- 本服务添加了一个代码生成器与文件直接关联的示例（Grid 布局生成器），其他生成器可自行拓展

- 保存快捷键与常用编辑器一致

- [Code Act Service](https://github.com/aisriver/code-act-service.git)

##### 遇到问题？或者你有更好的建议，欢迎一起讨论

- WeChat：mrliaojun Email：767882503@qq.com

### github

[Jared](https://github.com/aisriver/codect.git)


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

# [1.3.0](https://github.com/aisriver/codeact/compare/v1.2.0...v1.3.0) (2020-03-26)


### ✨ Features

* 添加git库 更新与表情 ([96dba19](https://github.com/aisriver/codeact/commit/96dba19)) by: **liaojun** (767882503@qq.com)


### 🐛 Bug Fixes

* fix version ([c9f6787](https://github.com/aisriver/codeact/commit/c9f6787)) by: **liaojun** (767882503@qq.com)


### 📝 Documentation

* update readme ([e93a8af](https://github.com/aisriver/codeact/commit/e93a8af)) by: **liaojun** (767882503@qq.com)



# [1.2.0](https://github.com/aisriver/codeact/compare/v1.0.4...v1.2.0) (2020-03-19)


### ✨ Features

* 新增从项目根目录读取/写入配置文件等功能 ([ce9aa90](https://github.com/aisriver/codeact/commit/ce9aa90)) by: **liaojun** (767882503@qq.com)


### 🐛 Bug Fixes

* @babel/runtime/regenerator bug fix ([579787a](https://github.com/aisriver/codeact/commit/579787a)) by: **liaojun** (767882503@qq.com)
* init 命令兼容windows ([2d0b487](https://github.com/aisriver/codeact/commit/2d0b487)) by: **liaojun** (767882503@qq.com)
* 移除无用注释 ([bb03176](https://github.com/aisriver/codeact/commit/bb03176)) by: **liaojun** (767882503@qq.com)



# [1.1.0](https://github.com/aisriver/codeact/compare/v1.0.4...v1.1.0) (2020-03-17)


### ✨ Features

* 新增从项目根目录读取/写入配置文件等功能 ([ce9aa90](https://github.com/aisriver/codeact/commit/ce9aa90)) by: **liaojun** (767882503@qq.com)



## [1.0.4](https://github.com/aisriver/codeact/compare/v1.0.3...v1.0.4) (2020-03-17)


### 🎫 Chores

* 手动引入runtime ([cad31e9](https://github.com/aisriver/codeact/commit/cad31e9)) by: **liaojun** (767882503@qq.com)



## [1.0.3](https://github.com/aisriver/codeact/compare/v1.0.2...v1.0.3) (2020-03-15)



## [1.0.2](https://github.com/aisriver/codeact/compare/v1.0.1...v1.0.2) (2020-03-15)


### 🎫 Chores

* add scripts ([9ff9de7](https://github.com/aisriver/codeact/commit/9ff9de7)) by: **liaojun** (767882503@qq.com)


### 🐛 Bug Fixes

* add babel-runtime ([48662d6](https://github.com/aisriver/codeact/commit/48662d6)) by: **liaojun** (767882503@qq.com)



## 1.0.1 (2020-03-15)


### 🎫 Chores

* init ([73a0208](https://github.com/aisriver/codeact/commit/73a0208)) by: **liaojun** (767882503@qq.com)


### 📝 Documentation

* add readme ([9285621](https://github.com/aisriver/codeact/commit/9285621)) by: **liaojun** (767882503@qq.com)





es/index.js
#! /usr/bin/env node
import _regeneratorRuntime from "@babel/runtime/regenerator";

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }

function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance"); }

function _iterableToArrayLimit(arr, i) { if (!(Symbol.iterator in Object(arr) || Object.prototype.toString.call(arr) === "[object Arguments]")) { return; } var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

import commander from 'commander';
import colors from 'colors';
import { logMessage, execPromise, getConfig } from './utils/process';
import { commandConfig } from './utils/command'; // 配置

var defaultConfig = {
  // code-act-service git地址
  gitAddress: 'https://github.com/aisriver/code-act-service.git',
  // 服务文件夹
  serviceFolderName: 'code-act-service',
  // 服务终止命令
  serviceStop: 'npm run stop',
  // 服务开始命令
  serviceStart: 'npm run start'
};
commander.version(require('../package.json').version).usage('[options] [project name]').option('-f, --find [value]', '查看service相关配置，gitAddress（服务端git地址）、serviceFolderName（服务文件夹名称）、serviceStop（服务终止命令）、serviceStart（服务开始命令）') //自定义帮助信息
.on('--help', function () {
  logMessage('\n\nCommands：', colors.white, '');
  logMessage('init        初始化\n', colors.green, '');
}).parse(process.argv);

var _commander$args = _slicedToArray(commander.args, 1),
    firstCommander = _commander$args[0];

var isInit = firstCommander === 'init';
/**
 * 进入服务目录
 */

var cdServiceDirectory = /*#__PURE__*/function () {
  var _ref = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee(config) {
    var gitAddress, serviceFolderName, result, createResult;
    return _regeneratorRuntime.wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            gitAddress = config.gitAddress, serviceFolderName = config.serviceFolderName;
            logMessage("\u51C6\u5907\u8FDB\u5165 ".concat(serviceFolderName, " \u6587\u4EF6\u5939"));
            _context.next = 4;
            return execPromise("cd ".concat(serviceFolderName));

          case 4:
            result = _context.sent;

            if (!result.error) {
              _context.next = 16;
              break;
            }

            // 如果没有就创建一个
            logMessage("\u672A\u53D1\u73B0 ".concat(serviceFolderName, " \u6587\u4EF6\u5939\uFF0C\u5F00\u59CB\u521B\u5EFA"), colors.red);
            _context.next = 9;
            return execPromise("git clone ".concat(gitAddress, " ").concat(serviceFolderName));

          case 9:
            createResult = _context.sent;

            if (!createResult.error) {
              _context.next = 13;
              break;
            }

            logMessage("\u521B\u5EFA\u670D\u52A1\u9047\u5230\u4E9B\u95EE\u9898\uD83D\uDE28\uFF1A".concat(createResult.error), colors.red);
            return _context.abrupt("return");

          case 13:
            logMessage("\u521B\u5EFA ".concat(serviceFolderName, " \u6210\u529F!\uD83D\uDE00"), colors.green);
            cdServiceDirectory(config);
            return _context.abrupt("return");

          case 16:
            logMessage("\u8FDB\u5165 ".concat(serviceFolderName, " \u6210\u529F!\uD83D\uDE03"), colors.green);
            startService(config);

          case 18:
          case "end":
            return _context.stop();
        }
      }
    }, _callee);
  }));

  return function cdServiceDirectory(_x) {
    return _ref.apply(this, arguments);
  };
}();
/**
 * 启动项目
 */


var startService = /*#__PURE__*/function () {
  var _ref2 = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee2(config) {
    var serviceFolderName, serviceStop, serviceStart, installResult, startResult;
    return _regeneratorRuntime.wrap(function _callee2$(_context2) {
      while (1) {
        switch (_context2.prev = _context2.next) {
          case 0:
            serviceFolderName = config.serviceFolderName, serviceStop = config.serviceStop, serviceStart = config.serviceStart;
            logMessage("\u66F4\u65B0/\u5B89\u88C5\u4F9D\u8D56\uFF0C\u8BF7\u8010\u5FC3\u7B49\u5F85".concat(isInit ? '（当前为初始化状态，时间可能会长一些😊）' : '😉', "..."));
            _context2.next = 4;
            return execPromise("cd ".concat(serviceFolderName, " && git stash && git pull && npm install"));

          case 4:
            installResult = _context2.sent;

            if (!installResult.error) {
              _context2.next = 8;
              break;
            }

            logMessage("\u66F4\u65B0/\u5B89\u88C5\u4F9D\u8D56\u9047\u5230\u4E9B\u95EE\u9898\uD83D\uDE2D\uFF1A".concat(installResult.error), colors.red);
            return _context2.abrupt("return");

          case 8:
            logMessage("\u4F9D\u8D56\u5B89\u88C5\u6210\u529F\uFF01\uD83D\uDE06", colors.green);
            logMessage("\u5F00\u59CB\u542F\u52A8\uD83D\uDE07...");
            _context2.next = 12;
            return execPromise("cd ".concat(serviceFolderName, " && ").concat(serviceStop));

          case 12:
            _context2.next = 14;
            return execPromise("cd ".concat(serviceFolderName, " && ").concat(serviceStart));

          case 14:
            startResult = _context2.sent;

            if (!startResult.error) {
              _context2.next = 18;
              break;
            }

            logMessage("\u670D\u52A1\u542F\u52A8\u9047\u5230\u4E9B\u95EE\u9898\uD83D\uDE23\uFF1A".concat(startResult.error), colors.red);
            return _context2.abrupt("return");

          case 18:
            logMessage("\u542F\u52A8\u6210\u529F\uFF01OPEN: http://localhost:7001", colors.green);

          case 19:
          case "end":
            return _context2.stop();
        }
      }
    }, _callee2);
  }));

  return function startService(_x2) {
    return _ref2.apply(this, arguments);
  };
}();
/**
 * 执行
 */


var codeAct = /*#__PURE__*/function () {
  var _ref3 = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee3() {
    var result, config, findValue, message;
    return _regeneratorRuntime.wrap(function _callee3$(_context3) {
      while (1) {
        switch (_context3.prev = _context3.next) {
          case 0:
            _context3.next = 2;
            return getConfig(defaultConfig);

          case 2:
            result = _context3.sent;

            if (result.isContinue) {
              _context3.next = 5;
              break;
            }

            return _context3.abrupt("return");

          case 5:
            config = result.config; // 配置查询

            if (!commander.find) {
              _context3.next = 10;
              break;
            }

            findValue = config[commander.find];

            if (findValue) {
              logMessage(findValue, colors.green, '');
            } else {
              message = "".concat(commander.find, "\u4E0D\u5728\u67E5\u8BE2\u8303\u56F4");

              if (typeof commander.find === 'boolean') {
                message = '请输入查询参数';
              }

              logMessage("".concat(message, "\uFF0C\u53EF\u67E5\u8BE2\u53C2\u6570\uFF08gitAddress | serviceFolderName | serviceStop | serviceStart\uFF09"), colors.red, '');
            }

            return _context3.abrupt("return");

          case 10:
            if (!isInit) {
              _context3.next = 14;
              break;
            }

            // 初始化状态先删除对应的文件夹
            logMessage("\u521D\u59CB\u5316...");
            _context3.next = 14;
            return execPromise("".concat(commandConfig["delete"], " ").concat(config.serviceFolderName));

          case 14:
            cdServiceDirectory(config);

          case 15:
          case "end":
            return _context3.stop();
        }
      }
    }, _callee3);
  }));

  return function codeAct() {
    return _ref3.apply(this, arguments);
  };
}();

codeAct();
//# sourceMappingURL=index.js.map
